### PR TITLE
add key vault secret and variable to api and web apps

### PR DESCRIPTION
**Pull request by Maby0**
# Pull Request Template

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] My code follows the style guidelines of this project
- [ ] My code changes do not include any hardcoded secrets or passwords
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My Tech Lead is aware of any infrastructure changes that will cost money

### CODE REVIEW

File: app/components/back-office-app-services/variables.tf [hunk 1]
```diff
@@ -208,6 +208,11 @@ variable "tags" {
   type        = map(string)
 }
 
+variable "feature_key_vault_enabled" {
```
Comment (chrisdalykinandcarta) [inline app/components/back-office-app-services/variables.tf:211]: Can we make this more descriptive? Something like "api_authorization_enabled" maybe? Bit confused about what it's for just reading the PR

Comment (Maby0) [reply app/components/back-office-app-services/variables.tf:211]: my only qualm with changing it to something like `api_authorization_enabled` is that even if that’s false, api auth checking is still technically happening, it’s just referencing a fixed test key that I’ve written here: https://github.com/Planning-Inspectorate/back-office/pull/2017/files#diff-43fca6de22bce10b10d2c788d167431d0ff290389e1953b2f76d0360a2c34905

Comment (chrisdalykinandcarta) [reply app/components/back-office-app-services/variables.tf:211]: That's a good point - maybe something like api_key_vault_authorization_enabled? I think feature_key_vault_enabled is maybe a bit too vague, and it implies that we're using Key Vault or not - and we use Key Vault for a lot of different things.

Comment (Maby0) [reply app/components/back-office-app-services/variables.tf:211]: Yeah that makes sense. Cool - `api_key_vault_authorization_enabled` sounds good. Will switch it up to that!

Comment (Maby0) [reply app/components/back-office-app-services/variables.tf]: addressed in latest commit

**Review by chrisdalykinandcarta — APPROVED** [review]

**Pull request closed without merging**